bundle/src/main/java/com/adobe/acs/commons/rewriter/impl/StylesheetInlinerTransformerFactory.java

```diff
@@ -0,0 +1,200 @@
+import org.apache.felix.scr.annotations.Deactivate;
+import org.apache.felix.scr.annotations.Properties;
+import org.apache.felix.scr.annotations.Property;
+import org.apache.felix.scr.annotations.Reference;
+import org.apache.felix.scr.annotations.Service;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.rewriter.ProcessingComponentConfiguration;
+import org.apache.sling.rewriter.ProcessingContext;
+import org.apache.sling.rewriter.Transformer;
+import org.apache.sling.rewriter.TransformerFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xml.sax.Attributes;
+import org.xml.sax.SAXException;
+import org.xml.sax.helpers.AttributesImpl;
+
+import com.adobe.acs.commons.rewriter.AbstractTransformer;
+import com.adobe.granite.ui.clientlibs.HtmlLibrary;
+import com.adobe.granite.ui.clientlibs.HtmlLibraryManager;
+import com.adobe.granite.ui.clientlibs.LibraryType;
+
+/**
+ * ACS AEM Commons - Stylesheet inliner removes stylesheet links the output adds
+ * them as <style> elements. Links found in <head> are added to the beginning of
+ * <body>, whereas those in <body> are included where they're found.
+ */
+@Component(
+        metatype = true, 
+        label = "Stylesheet Inliner Transformer Factory", 
+        description = "Sling Rewriter Transformer Factory which inlines CSS references")
+@Properties({ 
+    @Property( name = "pipeline.type", value = StylesheetInlinerTransformerFactory.SELECTOR, propertyPrivate = true) } )
+@Service(value = { TransformerFactory.class })
+public final class StylesheetInlinerTransformerFactory implements TransformerFactory {
+
+    public static final String SELECTOR = "inline-css";
+    
+    private static final String NEWLINE = "\n";
+    private static final String STYLE = "style";
+    private static final String BODY = "body";
+    private static final String HEAD = "head";
+
+    private static final Logger log = LoggerFactory.getLogger(StylesheetInlinerTransformerFactory.class);
+
+    @Reference
+    private HtmlLibraryManager htmlLibraryManager;
+
+
+    public Transformer createTransformer() {
+        return new CSSInlinerTransformer();
+    }
+
+    private class CSSInlinerTransformer extends AbstractTransformer {
+
+        protected boolean enabled = false;
+        protected boolean inHead = false;
+        protected List<String> stylesheetsInHead = new ArrayList<String>();
+        private SlingHttpServletRequest slingRequest;
+
+        @Override
+        public void init(ProcessingContext context, ProcessingComponentConfiguration config) throws IOException {
+            super.init(context, config);
+            slingRequest = context.getRequest();
+
+            String[] selectors = slingRequest.getRequestPathInfo().getSelectors();
+
+            if (selectors.length > 0) {
+                enabled = Arrays.asList(selectors).contains(SELECTOR);
```

Comment:
I'm not sure why this is necessary. Wouldn't you just this up in a separate pipeline which was triggered by the selector or just add this to the existing newsletter pipeline (default is in /libs/mcm/config/rewriter/campaign-link-rewrite)